rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergResponseObject'
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a table from the catalog
+      operationId: dropTable
+      description: Remove a table from the catalog
+      parameters:
+        - name: purgeRequested
+          in: query
+          required: false
+          description: Whether the user requested to purge the underlying table's data and metadata
+          schema:
+            type: boolean
+            default: false
+      responses:
+        '200':
+          description: OK
+          content:
+            'application/json':
+              schema:
+                type: boolean
+                description: true if the table was dropped and false if the table did not exist
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a table exists
+      operationId: tableExists
+      description:
+        Check if a table exists within a given namespace. Returns the standard response with `true` when found. Will return a TableNotFound error if not present.
+      parameters:
+        - name: namespace
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: table
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+        '412':
+          description: Table Not Found
+  /v1/namespaces/{namespace}/tables:
+    parameters:
+      - name: namespace
+        description: Namespace under which to list tables.
+        in: path
+        required: true
+        schema:
+          type: string
+        examples:
+          singlepart_namespace:
+            value: "accounting"
+          multipart_namespace:
+            value: "accounting.tax"
+    get:
+      tags:
+        - Catalog API
+      summary: List all table identifiers underneath a given namespace
+      description: Return all table identifiers under this namespace
+      operationId: listTables
+      parameters:
+        - name: namespace
+          description: Namespace under which to list tables.
+          in: path
+          required: true
+          schema:
+            type: string
+          examples:
+            singlepart_namespace:
+              value: "accounting"
+            multipart_namespace:
+              value: "accounting.tax"
+        # TODO - There's a much more native way to handle pagination
+        - name: limit
+          description: number of values to return in one request
+          in: query
+          required: false
+          schema:
+            type: integer
+          example: 10
+        - name: offset
+          description: Place in the response to continue from if paginating
+          in: query
+          required: false
+          schema:
+            type: integer
+      responses:
```

Comment:
There's a discussion elsewhere, but my concern with 404 is that in my experience, using 404 intentionally makes it hard to monitor a service (for example, many people monitor ELB response codes).

It's not possible from response code alone to disambiguate between expected 404 (e.g. namespace or other resouce doesn't exist, which isn't an operations concern generally speaking) and possibly misconfigured issues that people on-call would need to get paged for - like an unacceptable amount of 404s indicating they might need to check an nginx configuration or something to ensure that all the routes are being advertised appropriately.

There's still an open question of what response code to use instead, if any; I'm not entirely closed off to the idea to use 404.

But I have worked at places where our operations team eventually asked us to move off of intentional 404s so that they could monitor ELB response codes and get paged off of them without a fluctuating target for an acceptable number of 404s. And I have heard of similar efforts undertaken at Facebook etc, so I don't think it's unheard of to find using 404 undesirable.

Reply:
But yes that does need some sort of response code in it for namespace doesn't exist. Will add that. So far I've been using `412 - Precondition Failed` as namespace existing seem somewhat like a precondition that has failed.

I don't love it bu that's the rationale.

Reply:
Moved to 404.